compose/material3/material3/src/commonMain/kotlin/androidx/compose/material3/SearchBar.kt

```diff
@@ -457,7 +459,7 @@ private fun SearchBarInputField(
             .focusRequester(focusRequester)
             .onFocusChanged { if (it.isFocused) onActiveChange(true) }
             .onKeyEvent {
-                if (it.key == Key.Escape) {
+                if (it.key == Key.Escape && it.type == KeyEventType.KeyDown) {
```

Comment:
Not sure it's really needed, but just in case I've been doing it like this:

```
if (it.key == Key.Escape) {
    if (it.type == KeyEventType.KeyDown) {
        action()
    }
    true
}
else {
    false
}
```

So that they key-up is also consumed.

@igordmn What do you recommend?

Reply:
We do not consume up event, so we shouldn't fake-consume it

Reply:
I don't have a strong opinion here, but as I can't imagine cases where we need to handle KeyUp, even if KeyDown is consumed, I lean toward consuming everything.

Reply:
> We do not consume up event, so we shouldn't fake-consume it

This was my first response as well, but I didn't send it :). Let's Ivan decide.

Reply:
An argument for not consuming: 
- we should consume only what we handle

An argument for consuming:
- someone else can handle KeyUp, not knowing that another element consumes KeyDown

If we choose to consume, we can say that we consume "KeyDown gesture": KeyDown -> perform action -> KeyUp

Reply:
There are a list of places where we follow
> we should consume only what we handle

So, let's be consistent

Reply:
Yes, I agree, that we should do the same in this PR. the question - should we change the pattern everywhere.

Reply:
I think it's a bit safer to consume either both or neither. But since nobody has complained so far, maybe it doesn't affect (m)any people.